src/runner.jl

```diff
@@ -167,28 +175,66 @@ function test!(pkg::AbstractString,
-        cd(dirname(testfilepath)) do
-            try
-                cmd = ```
-                    $(Base.julia_cmd())
-                    --code-coverage=$(coverage ? "user" : "none")
-                    --color=$(Base.have_color ? "yes" : "no")
-                    --compiled-modules=$(Bool(Base.JLOptions().use_compiled_modules) ? "yes" : "no")
-                    --check-bounds=yes
-                    --depwarn=$(Base.JLOptions().depwarn == 2 ? "error" : "yes")
-                    --inline=$(Bool(Base.JLOptions().can_inline) ? "yes" : "no")
-                    --startup-file=$(Base.JLOptions().startupfile != 2 ? "yes" : "no")
-                    --track-allocation=$(("none", "user", "all")[Base.JLOptions().malloc_log + 1])
-                    $(runner_file_path)
-                    ```
-                run(cmd)
-                @info "$pkg tests passed. Results saved to $logfilename."
-            catch err
-                @warn "ERROR: Test(s) failed or had an error in $pkg"
-                push!(errs,pkg)
+        testreportsdir = dirname(@__DIR__)
+        runner_code = gen_runner_code(testfilepath, logfilename, testreportsdir)
+        cmd = ```
+            $(Base.julia_cmd())
+            --code-coverage=$(coverage ? "user" : "none")
+            --color=$(Base.have_color ? "yes" : "no")
+            --compiled-modules=$(Bool(Base.JLOptions().use_compiled_modules) ? "yes" : "no")
+            --check-bounds=yes
+            --depwarn=$(Base.JLOptions().depwarn == 2 ? "error" : "yes")
+            --inline=$(Bool(Base.JLOptions().can_inline) ? "yes" : "no")
+            --startup-file=$(Base.JLOptions().startupfile != 2 ? "yes" : "no")
+            --track-allocation=$(("none", "user", "all")[Base.JLOptions().malloc_log + 1])
+            --eval $(runner_code)
+            ```
+
+        test_folder_has_project_file = isfile(replace(testfilepath, "runtests.jl"=>"Project.toml"))
+
+        if VERSION >= v"1.4.0" || (VERSION >= v"1.2.0" && test_folder_has_project_file)
+            # Operations.sandbox() has different arguments between versions
+            @static if VERSION >= v"1.4.0"
+                test_project_override = test_folder_has_project_file ?
+                    nothing :
+                    gen_target_project(ctx, pkgspec, pkgspec.path, "test")
+                sandbox_args = (ctx,
+                                pkgspec,
+                                pkgspec.path,
+                                joinpath(pkgspec.path, "test"),
+                                test_project_override)
+            else
+                sandbox_args = (ctx,
+                                pkgspec,
+                                pkgspec.path,
+                                joinpath(pkgspec.path, "test"))
+            end
+
+            sandbox(sandbox_args...) do
+                # test_fn !== nothing && test_fn()  # TODO: implement
+                flush(stdout)
```

Comment:
Is there a reason this is needed?
Normally it self-flushes just fine.

Reply:
Not sure, I was copying it directly from [this line](https://github.com/JuliaLang/Pkg.jl/blob/master/src/Operations.jl#L1542) from `Pkg.jl/src/Operations.jl`:

Can definitely remove if you think its not needed?